internal/workflow/processing.go

```diff
@@ -803,51 +772,98 @@ func (w *ProcessingWorkflow) waitForReview(ctx temporalsdk_workflow.Context) *pa
+	cleanup *cleanupRegistry,
+) error {
+	// Bundle PIP as an Archivematica standard transfer.
+	{
+		activityOpts := withActivityOptsForLongLivedRequest(sessCtx)
+		var bundleResult activities.BundleActivityResult
+		err := temporalsdk_workflow.ExecuteActivity(
+			activityOpts,
+			activities.BundleActivityName,
+			&activities.BundleActivityParams{
+				SourcePath:  tinfo.TempPath,
+				TransferDir: w.cfg.A3m.ShareDir,
+				IsDir:       tinfo.IsDir,
+			},
+		).Get(activityOpts, &bundleResult)
+		if err != nil {
+			return err
+		}
 
-	params := &a3m.CreateAIPActivityParams{
-		Name:                 tinfo.Name(),
-		Path:                 tinfo.Bundle.FullPath,
-		PreservationActionID: tinfo.PreservationActionID,
-	}
+		tinfo.Bundle = bundleResult
+		tinfo.PackageType = enums.PackageTypeArchivematicaStandardTransfer
 
-	result := a3m.CreateAIPActivityResult{}
-	err := temporalsdk_workflow.ExecuteActivity(activityOpts, a3m.CreateAIPActivityName, params).Get(sessCtx, &result)
-	if err != nil {
-		return err
+		// Delete bundled transfer when session ends.
+		cleanup.registerPath(bundleResult.FullPath)
 	}
 
-	tinfo.SIPID = result.UUID
-	tinfo.AIPPath = result.Path
-	tinfo.StoredAt = temporalsdk_workflow.Now(sessCtx).UTC()
+	// Send PIP to a3m for preservation.
+	{
+		activityOpts := temporalsdk_workflow.WithActivityOptions(sessCtx, temporalsdk_workflow.ActivityOptions{
+			StartToCloseTimeout: time.Hour * 24,
+			HeartbeatTimeout:    time.Second * 5,
+			RetryPolicy: &temporalsdk_temporal.RetryPolicy{
+				MaximumAttempts: 1,
+			},
+		})
+
+		params := &a3m.CreateAIPActivityParams{
+			Name:                 tinfo.Name(),
+			Path:                 tinfo.Bundle.FullPath,
+			PreservationActionID: tinfo.PreservationActionID,
+		}
+
+		result := a3m.CreateAIPActivityResult{}
+		err := temporalsdk_workflow.ExecuteActivity(activityOpts, a3m.CreateAIPActivityName, params).
+			Get(sessCtx, &result)
+		if err != nil {
+			return err
+		}
+
+		tinfo.SIPID = result.UUID
+		tinfo.AIPPath = result.Path
+		tinfo.StoredAt = temporalsdk_workflow.Now(sessCtx).UTC()
+	}
 
 	return nil
 }
 
-func (w *ProcessingWorkflow) transferAM(sessCtx temporalsdk_workflow.Context, tinfo *TransferInfo) error {
+func (w *ProcessingWorkflow) transferAM(ctx temporalsdk_workflow.Context, tinfo *TransferInfo) error {
 	var err error
 
-	// Zip transfer.
-	activityOpts := withActivityOptsForLongLivedRequest(sessCtx)
+	// Bag PIP if it's not already a bag.
+	if tinfo.PackageType != enums.PackageTypeBagIt {
+		lctx := withActivityOptsForLocalAction(ctx)
+		var zipResult bagit_activity.CreateBagActivityResult
+		err = temporalsdk_workflow.ExecuteActivity(
+			lctx,
+			bagit_activity.CreateBagActivityName,
+			&bagit_activity.CreateBagActivityParams{SourcePath: tinfo.TempPath},
+		).Get(lctx, &zipResult)
+		if err != nil {
+			return err
+		}
+		tinfo.PackageType = enums.PackageTypeBagIt
+	}
+
+	// Zip PIP.
+	activityOpts := withActivityOptsForLongLivedRequest(ctx)
```

Comment:
Should we use `withActivityOptsForLocalAction(ctx)` instead of `withActivityOptsForLongLivedRequest(ctx)` for the zip activity too?

Reply:
Yeah, good idea. I can't imagine zipping a PIP to take longer than an hour. :sweat_smile: